client/src/test/java/org/eclipse/hono/client/impl/HonoClientUnitTestHelper.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 1.0 which is available at
+ * https://www.eclipse.org/legal/epl-v10.html
+ *
+ * SPDX-License-Identifier: EPL-1.0
+ */
+
+package org.eclipse.hono.client.impl;
+
+import io.vertx.core.Context;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.proton.ProtonReceiver;
+import io.vertx.proton.ProtonSender;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Helper class that provides mocks for objects that are needed by unit tests using a Hono client.
+ *
+ */
+public final class HonoClientUnitTestHelper {
+
+    private HonoClientUnitTestHelper() {}
+
+    /**
+     * Create a mock for the Vert.x context that is used for the unit test.
```

Comment:
I like that you added JavaDocs :+1: 
They would be even more helpful, if they would describe the returned mock's behavior, e.g. in this case something like
> Creates a mocked vert.x *Context* which immediately invokes any handler that is passed to its *runOnContext* method.

Otherwise, any user of these methods will still need to look into the source code in order to be able to use any of the methods properly.

Reply:
Fully agree - will improve the description!